lib/vyasa_web/components/contexts/read.ex

```diff
@@ -716,31 +699,54 @@ defmodule VyasaWeb.Context.Read do
         } = _params,
         %Socket{
           assigns: %{
+            marks_ui: %MarksUiState{} = ui_state,
             reply_to: %Sheaf{} = parent_sheaf,
-            draft_reflector: %Sheaf{},
+            draft_reflector:
+              %Sheaf{
+                id: draft_id,
+                marks: marks
+              } = _draft_reflector,
             session: %VyasaWeb.Session{
               name: user_signature,
-              sangh: sangh
+              sangh:
+                %Vyasa.Sangh.Session{
+                  id: sangh_session_id
+                } = _sangh
             }
           }
-          # reply_to:  %Sheaf{id: _parent_id} = parent_sheaf
         } = socket
-      ) do
+      )
+      when not is_nil(parent_sheaf) do
     IO.inspect(%{body: body, is_private: is_private},
       label: "SHEAF CREATION"
     )
 
-    Vyasa.Sangh.create_sheaf(%{
-      id: Ecto.UUID.generate(),
-      body: body,
-      session_id: sangh.id,
-      parent: parent_sheaf,
-      signature: user_signature
-    })
+    created_sheaf =
+      Vyasa.Sangh.create_sheaf(%{
+        id: draft_id || Ecto.UUID.generate(),
+        active: true,
+        body: body,
+        marks: marks,
+        traits: ["published"],
+        session_id: sangh_session_id,
+        parent: parent_sheaf,
+        signature: user_signature
```

Comment:
this is what the payload should be when updating the sheaf

it's more of an upsert() and this is the payload